src/helpers/files.js

```diff
@@ -177,19 +177,21 @@ const addMenuRichWorkspace = () => {
 			})
 			const fileid = parseInt(response.headers['oc-fileid'])
 			const file = new File({
-				source,
+				source: context.source + '/' + encodeURIComponent(descriptionFile),
 				id: fileid,
 				mtime: new Date(),
 				mime: 'text/markdown',
 				owner: getCurrentUser()?.uid || null,
 				permissions: Permission.ALL,
 				root: context?.root || '/files/' + getCurrentUser()?.uid,
 			})
-			context._children.push(file.fileid)
 
 			showSuccess(t('text', 'Created "{name}"', { name: descriptionFile }))
+
 			emit('files:node:created', file)
-			emit('Text::showRichWorkspace', { autofocus: true })
+			setTimeout(() => {
+				emit('Text::showRichWorkspace', { autofocus: true })
+			}, 200)
```

Comment:
Can you tell why the previous one didn't work?

Reply:
When user directly accessed an empty folder, the render() (at https://github.com/nextcloud/text/blob/main/src/helpers/files.js#L207) did not run. So, RichWorkspace component was not created.
Then, triggering `Text::showRichWorkspace` right after creating Readme.md did not work.
The RichWorkspace component was created only after any file or folder was created.
So, we have to wait some time until RichWorkspace component is created.

Reply:
Ah i see, that seems tricky. Let's get this in then for now. We might want to have the files app adjusted so that the header component itself is also rendered for empty file lists mid term. I only pushed a commit to use the file created event instead which seemed more clean.